scipy-1.0/poly.tex

```diff
@@ -28,23 +16,12 @@
 \code{PPoly} represents piecewise polynomials in the power basis in terms of
 breakpoints and coefficients at each interval. \code{BPoly} is similar, and
 represents piecewise polynomials in the Bernstein basis (which is suitable
-for e.g., constructing Bezier curves).
-
-In the next layer, these polynomial classes are used for constructing several common ways of interpolating data: \code{CubicSpline} constructs a twice differentiable piecewise cubic function, \code{Akima1DInterpolator} and \code{PCHIPInterpolator} implement two classic prescriptions for constructing a $C^1$ continuous monotone shape-preserving interpolator. \cite{FritschCarlson1980, Akima1970}
-%
-Interpolation with known or estimated derivatives, i.e., cubic or higher-order Hermite splines, can be constructed by either subclassing \code{PPoly} or \code{BPoly}, or using the factory method \code{BPoly.from\_derivatives}.
-
-For working with b-splines, there are several options. The basic object is \code{BSpline}, which represents spline curves, i.e., linear combinations of b-spline basis elements.\cite{deBoor1978}. The spline
-representation used by \code{BSpline} is directly compatible with FITPACK. We
-recommend that users work with \code{BSpline} objects instead of directly
-manipulating \code{tck} tuples of knots, coefficients and the spline degree.
-%
-Using \code{UnivariateSpline} in new code is not recommended. Given data points,
-the helper function \code{make\_interp\_spline} constructs a \code{BSpline}
-instance, representing a spline which interpolates the data. Interpolated values
-at intermediate points can be computed either via its \code{\_\_call\_\_} method,
-or by calling \code{splev(xnew, BSpline\_object)} --- the \code{splev} routine
-accepts both \code{tck} tuples and \code{Bspline} instances.
-
-Fitting \emph{smoothing} splines to data can currently only be done with FITPACK. Building alternative algorithms is certainly possible, but is still to be implemented.
+for e.g., constructing Bezier curves). \code{BSpline} represents spline
```

Comment:
@ev-br Could you give some dates and / or PR links for these new things added in last 3 years or so?

Reply:
PPoly & BPoly went in in https://github.com/scipy/scipy/pull/5827  for scipy 0.13
BSpline went in in scipy/scipy#3174 for scipy 0.19
NdPPoly went in for scipy/scipy#5827 for scipy 0.18
(Now that I think of it, it is worth mentioning here, if briefly)
CubicSpline was implemented in scipy/scipy#5653, new in scipy 0.18

Reply:
I added in the first two & the last one. I'm confused about the first one though -- it is the same as the third, but for a different version of SciPy.

Reply:
That's a typo, sorry. Edited in the correct PR, which is https://github.com/scipy/scipy/pull/2885

Reply:
Revised to point to Pauli's PR, thanks